BLUESPAWN-client/src/hunt/hunts/HuntT1101.cpp

```diff
@@ -16,49 +16,34 @@ namespace Hunts {
 		dwTacticsUsed = (DWORD) Tactic::Persistence;
 	}
 
-	int HuntT1101::EvaluatePackages(Registry::RegistryKey key, std::vector<std::wstring> vSecPackages, Reaction reaction) {
-		int detections = 0;
-
-		FileSystem::File* file;
-
-		for (auto secPackage : vSecPackages) {
-			if (secPackage == L"\"\"") {
-				continue;
-			}
-			if (FileSystem::File(secPackage).GetFileExists()) {
-				file = new FileSystem::File(secPackage);
-			}
-			else {
-				file = new FileSystem::File(ExpandEnvStringsW(L"%SYSTEMROOT%\\System32\\") + secPackage + L".dll");
-				if (!file->GetFileExists()) { continue; }
-			}
-
-			auto& yara = YaraScanner::GetInstance();
-			YaraScanResult result = yara.ScanFile(*file);
-
-			if (!file->GetFileSigned()) {
-				reaction.RegistryKeyIdentified(std::make_shared<REGISTRY_DETECTION>(RegistryValue{ key, L"Security Packages", key.GetValue<std::vector<std::wstring>>(L"Security Packages").value() }));
-				reaction.FileIdentified(std::make_shared<FILE_DETECTION>(file->GetFilePath()));
-				detections += 2;
-			}
-		}
-
-		return detections;
-	}
-
 	int HuntT1101::ScanCursory(const Scope& scope, Reaction reaction){
 		LOG_INFO(L"Hunting for " << name << " at level Cursory");
 		reaction.BeginHunt(GET_INFO());
 
 		int detections = 0;
 
 		auto lsa = RegistryKey{ HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Control\\Lsa" };
-		auto vSecPackages = *lsa.GetValue<std::vector<std::wstring>>(L"Security Packages");
-		detections += EvaluatePackages(lsa, vSecPackages, reaction);
-
 		auto lsa2 = RegistryKey(HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Control\\Lsa\\OSConfig");
-		auto vSecPackages2 = *lsa2.GetValue<std::vector<std::wstring>>(L"Security Packages");
-		detections += EvaluatePackages(lsa2, vSecPackages2, reaction);
+
+		for (auto key : { lsa, lsa2 }) {
+			auto Packages = key.GetValue<std::vector<std::wstring>>(L"Security Packages");
+			if (Packages) {
+				for (auto Package : Packages.value()) {
+					if (Package != L"\"\"") {

```

Comment:
Here, it looks like you're checking for an empty set of quotes, but in the next statement, you assume the value is not quoted. Is this a valid assumption?